prometheus-metrics/src/main/scala/org/http4s/metrics/prometheus/Prometheus.scala

```diff
@@ -191,6 +193,74 @@ object Prometheus {
   private val DefaultHistogramBuckets: NonEmptyList[Double] =
     NonEmptyList(.005, List(.01, .025, .05, .075, .1, .25, .5, .75, 1, 2.5, 5, 7.5, 10))
 
+  /**
+    * Given an exclude function, return a 'classifier' function, i.e. for application in
+    * [[org.http4s.server/client.middleware.Metrics#apply]].
+    *
+    * Note that the output String will be lower-cased.
+    *
+    * Let's say you want a classifier that excludes integers since your paths consist of:
+    *   * GET    /users/{integer} = get_users_*
+    *   * POST   /users           = post_users
+    *   * PUT    /users/{integer} = put_users_*
+    *   * DELETE /users/{integer} = delete_users_*
+    *
+    * In such a case, we could use:
+    *
+    * classifierFMethodWithOptionallyExcludedPath(
+    *   exclude          = { str: String => scala.util.Try(str.toInt).isSuccess },
+    *   excludedValue    = "*",
+    *   intercalateValue = "_"
+    * )
+    *
+    * @param exclude For a given String, namely a path value, determine whether the value gets excluded.
+    * @param excludedValue Indicates the String value to be supplied for an excluded path's field.
+    * @param pathSeparator Value to use for separating the metrics fields' values
+    * @return Request[F] => Option[String]
+    */
+  def classifierFMethodWithOptionallyExcludedPath[F[_]](
+      exclude: String => Boolean,
```

Comment:
Perhaps worth noting that this run on every segment of a path. So note that if an intermediate throwables with Stack traces is known and discarded there may be a performance penalty, like the examples you have shown with `Int` and `UUID`, both I benchmarked some approaches and regex matches should generally be preferred over throwables in this position.

Reply:
Thanks for pointing that out. I added https://github.com/http4s/http4s/pull/3167/commits/48c0661ca6a6ce800d39332685139a9b95c80541.